File: src/main/java/com/github/nsc/de/compiler/parser/node/expression/DoubleNode.java
package com.github.nsc.de.compiler.parser.node.expression;

import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class DoubleNode implements ValuedNode {

    private final double number;

    public DoubleNode(double number) {
        this.number = number;
    }

    public double getNumber() {
        return number;
    }

    @Override
    public String toString() {
        return number+"";
    }
}

File: parser/src/commonMain/kotlin/com/github/nsc/de/shake/parser/node/factor/DoubleNode.kt
package com.github.nsc.de.shake.parser.node.factor

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class DoubleNode(map: PositionMap, val number: Double) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "DoubleNode", "value" to number)
}